sn_client/src/api.rs

```diff
@@ -419,16 +431,31 @@ impl Client {
         }
     }
 
-    /// get_store_cost_for_dbc_id
-    pub async fn get_store_cost_for_dbc_id(&self, dbc_id: &DbcId) -> Result<Token> {
-        let address = DbcAddress::from_dbc_id(dbc_id);
-        let net_address = NetworkAddress::from_dbc_address(address);
-
-        trace!("Getting store cost at {dbc_id:?}");
+    /// Get the store cost at a given address
+    /// Replaces current network_store_cost with the new one, unless average is set to true
+    pub async fn get_store_cost_at_address(
+        &mut self,
+        address: NetworkAddress,
+        average_cost: bool,
+    ) -> Result<Token> {
+        trace!("Getting store cost at {address:?}");
 
-        Ok(self
+        let cost = self
             .network
-            .get_store_cost_from_network(net_address.clone())
-            .await?)
+            .get_store_cost_from_network(address.clone())
+            .await?
+            .as_nano();
+
+        if average_cost {
+            if let Some(current_cost) = self.network_store_cost {
+                self.network_store_cost = Some((current_cost + cost) / 2);
+            } else {
+                self.network_store_cost = Some(cost);
+            }
+        } else {
+            self.network_store_cost = Some(cost);
```

Comment:
if not averaging, shall we consider only update when the incoming cost is higher ?

Reply:
I don't think so? The price could reduce as much as go up?

Reply:
well, it's bit tricky here as I understand.
This storage_cost will be used for a payment targeting any part of the network.
If talking about a snapshot of the whole network's storage cost distribution at specific time, it shall always be the highest one to prefer, to secure the cost we pay can be accepted by any part of the network.

However, as you said, if talking about the storage cost along the time, it does can go down as well.

In general, I feel picking the highest one is the safest way to ensure the payment can be accepted across the network, within the current mechanism of how storage cost got calculated by client and validated by network nodes on receiving a payment.

Reply:
Ahhh, yeh, i totally agree with what you're saying. I thought that's what we were doing here but it's not really.

We're averaging, which means we won't pay to a part of the network. So we actually should not avg, but just increase if higher.